auto_inertial_params/proposal.md

```diff
@@ -0,0 +1,410 @@
+
+ 2. `//collision/density` element for specifying the density of the collision geometry of a link. 
+ This density values would be used to calculate the inertial parameters of the respective collision 
+ geometries. Adding this as part of the `<collision>` tag would allow a user to simulate links with 
+ different material types for different collisions. By default, the value of density would be set 
+ equal to that of water which is 1000 kg/m^3. A `//link/inertial/density` element would also be 
+ added in the spec to allow users to specify the density values on a link level instead of 
+ specifying the same values for each collision.
+
+ 3. `//collision/auto_inertia_params` element would be added that can be used to provide some 
+ parameters or options for a custom inertia calculator. Similar to the `density` element above, 
+ a link-level `//link/inertial/auto_inertia_params` element would also be added. This would allow 
+ the user to provide inertia calculator parameters for the whole link while retaining the 
+ flexibility to specify different parameters for each collision. Custom elements and attributes 
+ using a namespace prefix can be used to provide the user-defined parameters for a custom calculator. 
+ More about this can be found in [this](http://sdformat.org/tutorials?tut=custom_elements_attributes_proposal&cat=pose_semantics_docs&#specifying-custom-elements-and-attributes) proposal.
+
+The example snippet below shows how the above proposed elements would be used in a SDFormat `<link>`:
+
+```xml
+<link name="robot_link">
+  <inertial auto="true">
+    <auto_inertia_params>
+      <gz:voxel_size>0.01</gz:voxel_size>
+    </auto_inertia_params>
+  <collision name="collision">
+    <density>*some_float_value*</density>
+    <geometry>
+      .
+      .
+    </geometry>
+  </collision>
+  <visual name="visual">
+    <geometry>
+      .
+      .
+    </geometry>
+    <material>
+      .
+      .
+    </material>
+  </visual>
+</link>
+```
+
+## Proposed Implementation
+> Note: In the section below, the term **primitive geometries** is used to collectively 
+describe Box, Capusle, Cylinder, Ellipsoid and Sphere geometries.
+
+### Some Architectural Considerations
+
+Below are some key architectural considerations for the implementation of this feature:
+
+ *  The parsing of the proposed SDFormat elements and the Moment of Inertia calculations 
+ for primitive geometries (Box, Cylinder, Sphere, Ellipsoid and Capsule) can be implemented 
+ as an integral part of `libsdformat`. This would help enable all simulators that rely on 
+ SDFormat to utilize this feature.
+
+ * In case of 3D meshes being used as geometries, a modular callback-based architecture 
+ can be followed where the user can integrate their custom Moments of Inertia calculator. 
+ [Two methods](#proposed-mesh-inertia-calculation-methods) were explored computing the 
+ inertial properties (mass, mass matrix and center of mass) of 3D meshes: a [Voxelization-based] 
```

Comment:
```suggestion
 inertial properties (mass, mass matrix and center of mass) of 3D meshes: a [Voxelization-based](#voxelization-based-method)
```

Reply:
Corrected in [fca06ae](https://github.com/gazebosim/sdf_tutorials/pull/92/commits/fca06ae64a6843311fa125ff89c6f9a376c37599)